### PR TITLE
[YUNIKORN-1528] Add gang scheduling test suites - Phase 2

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?
Cover the left out test methods of Gang scheduling test suite.

### What type of PR is it?
* [ ] - Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1528

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by wilfred-s** [comment]
@manirajv06 we probably need YUNIKORN-1525 fixed so we can get the unit tests to pass and run the e2e tests.
I kicked of the build again and see if it at least gets us passed the unit tests now.

File: test/e2e/gang_scheduling/gang_scheduling_test.go [hunk 1]
```diff
@@ -520,6 +521,259 @@ var _ = Describe("", func() {
...
+			PodConfig:   podConf,
+		}
+
+		// Create gang job
+		job, jobErr := k8s.InitJobConfig(jobConf)
+		Ω(jobErr).NotTo(HaveOccurred())
+		taskGroupsMap, annErr := k8s.PodAnnotationToMap(podConf.Annotations)
+		Ω(annErr).NotTo(HaveOccurred())
+		By(fmt.Sprintf("[%s] Deploy job %s with task-groups: %+v",
+			podConf.Labels["applicationId"], jobConf.Name, taskGroupsMap[k8s.TaskGroups]))
```
Comment (wilfred-s) [inline test/e2e/gang_scheduling/gang_scheduling_test.go:718]: nit layout needs cleanup: should be one line we have no length limit

File: test/e2e/gang_scheduling/gang_scheduling_test.go [hunk 2]
```diff
@@ -520,6 +521,259 @@ var _ = Describe("", func() {
...
+			apps[0]: {"pods": 1, "?": 0},
+			apps[1]: {"pods": 3, "minMembers": 3},
+			apps[2]: {"pods": 1, "minMembers": 0},
+		}
+		// Expected appState progression
+		appStates := map[string][]string{
+			apps[0]: {"Starting", "Completing"},
+			apps[1]: {"Accepted", "Running"},
+			apps[2]: {"Accepted", "Accepted"},
+		}
```
Comment (wilfred-s) [inline test/e2e/gang_scheduling/gang_scheduling_test.go:558]: As we have discussed: we need to be 100% sure that these states are stable. Please document why we expect these states and why they are stable

**Review by wilfred-s — CHANGES_REQUESTED** [review]
mostly layout changes

File: test/e2e/gang_scheduling/gang_scheduling_test.go [hunk 3]
```diff
@@ -31,12 +31,18 @@ import (
...
 
 var _ = Describe("", func() {
 	var kClient k8s.KubeCtl //nolint
 	var restClient yunikorn.RClient
 	var ns string
+	groupA := "groupa"
+	groupB := "groupb"
+	fifoQName := "fifoq"
+	defaultPartition := "default"
+	rootNamespace := "root." + ns
```
Comment (wilfred-s) [inline test/e2e/gang_scheduling/gang_scheduling_test.go:45]: This is the queue based on the namespace (ns), would expect it to be called something like `nsQueue` or just `queue`. `ns` gets set in BeforeEach(). Does that not mean the value is off which causes the e2e failures?

**Comment by manirajv06** [comment]
Cleaned up all debug statements. All tests are passing consistently, ready for merge. Raised https://issues.apache.org/jira/browse/YUNIKORN-1533 for v1.22.15 failures. It is related to the untrusted users.

File: test/e2e/framework/helpers/yunikorn/rest_api_utils.go [hunk 4]
```diff
@@ -404,12 +406,13 @@ func (c *RClient) LogNodesInfo(outputDir string) error {
...
-func (c *RClient) GetPartitions() (*dao.PartitionDAOInfo, error) {
-	req, err := c.newRequest("GET", configmanager.QueuesPath, nil)
+func (c *RClient) GetPartitions(partition string) (*dao.PartitionQueueDAOInfo, error) {
+	req, err := c.newRequest("GET", fmt.Sprintf(configmanager.QueuesPath, partition), nil)
 	if err != nil {
 		return nil, err
 	}
-	partitions := new(dao.PartitionDAOInfo)
-	_, err = c.do(req, partitions)
+	// partitions := new(dao.PartitionDAOInfo)
```
Comment (pbacsko) [inline test/e2e/framework/helpers/yunikorn/rest_api_utils.go:414]: Nit: unnecessary comment

**Comment by pbacsko** [comment]
+1 merging soon

**Pull request closed without merging**